src/apps/explorer/components/TokensTableWidget/BatchesTableWithData.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useContext, useState, useEffect } from 'react'
+
+import { EmptyItemWrapper } from 'components/common/StyledUserDetailsTable'
+import useFirstRender from 'hooks/useFirstRender'
+import { TokensTableContext } from 'apps/explorer/components/TokensTableWidget/context/TokensTableContext'
+import BatchesTable from 'components/batches/BatchesTable'
+import { DEFAULT_TIMEOUT } from 'const'
+import CowLoading from 'components/common/CowLoading'
+
+export const BatchesTableWithData: React.FC = () => {
+  const { batchesData: batches, networkId, tableState } = useContext(TokensTableContext)
+  const isFirstRender = useFirstRender()
+  const [isFirstLoading, setIsFirstLoading] = useState(true)
+
+  useEffect(() => {
+    setIsFirstLoading(true)
+  }, [networkId])
+
+  useEffect(() => {
+    let timeOutMs = 0
+    if (!batches) {
+      timeOutMs = DEFAULT_TIMEOUT
+    }
+
+    const timeOutId: NodeJS.Timeout = setTimeout(() => {
+      setIsFirstLoading(false)
+    }, timeOutMs)
```

Comment:
these logics seems overcomplicated, plus is strange that you are using this for the loading, but is only based on a timer... 

Wouldn't be cleaner design if the context would have a `loading: boolean` flag ? so responsibility is not in this component (the loading logic is not here anyways, so it doesn't feel is this component responsibility)

Reply:
That's the same logic used to load the Tokens too